File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int count[] = new int[nums.length+1]
        ;
        for(int x:nums)
        {
            count[x]++;
        }
        for(int i=0;i<count.length;i++)
        {
            if(count[i]==0)return i;
        }

        return 0;
    }
}

File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {

        int missingNum = 0;
        for(int i =0  ; i<nums.size() ; i++)
        {
            missingNum^=nums[i];

        }

        for(int i=0 ; i<nums.size()+1 ; i++)
        {
            missingNum^=i;

        }

        return missingNum;


        
    }
};